### PR TITLE
All layers with raster cache should use integer CTM (#33981)

### PR DESCRIPTION
**Pull request by jonahwilliams**
Cherry pick for https://github.com/flutter/flutter/issues/106523 . Manually verified with the example code that the artifacts are no longer present

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I had to change a lot more due to some churn from a separate layer caching change, so now this needs additional verification from me

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-engine-gold.skia.org/cl/github/34385

**Review by CaseyHillers — APPROVED** [review]

**Review by zanderso — APPROVED** [review]

File: flow/layers/checkerboard_layertree_unittests.cc [hunk 1]
```diff
@@ -59,9 +59,6 @@ TEST_F(CheckerBoardLayerTest, ClipRectSaveLayerNotCheckBoard) {
            MockCanvas::DrawCall{
                1, MockCanvas::ClipRectData{layer_bounds, SkClipOp::kIntersect,
                                            MockCanvas::kSoft_ClipEdgeStyle}},
-#ifndef SUPPORT_FRACTIONAL_TRANSLATION
```
Comment (jonahwilliams) [inline flow/layers/checkerboard_layertree_unittests.cc]: These were removed since the code to raster cache clips hasn't been added yet

**Comment by skia-gold** [comment]
Gold has detected about 8 new digest(s) on patchset 2.
View them at https://flutter-engine-gold.skia.org/cl/github/34385

**Pull request merged**